src/tuntap/windows/c_tuntap_windows.cpp

```diff
@@ -0,0 +1,297 @@
+	return 0;
+}
+
+size_t c_tuntap_windows_obj::read_from_tun(unsigned char *const data, size_t size) {
+	return 0;
+}
+
+void c_tuntap_windows_obj::async_receive_from_tun(unsigned char *const data, size_t size, const c_tuntap_base_obj::read_handler &handler) {
+}
+
+void c_tuntap_windows_obj::set_tun_parameters(const std::array<unsigned char, 16> &binary_address, int prefix_len, uint32_t mtu) {
+	_fact("Setting IPv6 address, prefixLen=" << prefix_len);
+	std::wstring human_name = get_human_name(m_guid);
+}
+
+
+// base on https://msdn.microsoft.com/en-us/library/windows/desktop/ms724256(v=vs.85).aspx
+std::vector<std::wstring> c_tuntap_windows_obj::get_subkeys(HKEY hKey) {
+	constexpr size_t mex_key_length = 255;
+	constexpr size_t max_value_name = 16383;
+	TCHAR    achKey[mex_key_length];   // buffer for subkey name
+	DWORD    cbName;                   // size of name string
+	TCHAR    achClass[MAX_PATH] = TEXT("");  // buffer for class name
+	DWORD    cchClassName = MAX_PATH;  // size of class string
+	DWORD    cSubKeys = 0;               // number of subkeys
+	DWORD    cbMaxSubKey;              // longest subkey size
+	DWORD    cchMaxClass;              // longest class string
+	DWORD    cValues;              // number of values for key
+	DWORD    cchMaxValue;          // longest value name
+	DWORD    cbMaxValueData;       // longest value data
+	DWORD    cbSecurityDescriptor; // size of security descriptor
+	FILETIME ftLastWriteTime;      // last write time
+	DWORD retCode;
+	std::vector<std::wstring> ret;
+	TCHAR  achValue[max_value_name];
+	DWORD cchValue = max_value_name;
+
+	// Get the class name and the value count.
+	_fact("Query windows registry for infokeys");
+	retCode = RegQueryInfoKey(
+		hKey,                    // key handle
+		achClass,                // buffer for class name
+		&cchClassName,           // size of class string
+		nullptr,                 // reserved
+		&cSubKeys,               // number of subkeys
+		&cbMaxSubKey,            // longest subkey size
+		&cchMaxClass,            // longest class string
+		&cValues,                // number of values for this key
+		&cchMaxValue,            // longest value name
+		&cbMaxValueData,         // longest value data
+		&cbSecurityDescriptor,   // security descriptor
+		&ftLastWriteTime);       // last write time
+								 // Enumerate the subkeys, until RegEnumKeyEx fails.
+	if (retCode != ERROR_SUCCESS) throw std::runtime_error("RegQueryInfoKey error, error code " + std::to_string(GetLastError()));
+	if (cSubKeys > 0) {
+		_fact("Number of subkeys: " << cSubKeys);
+
+		for (DWORD i = 0; i < cSubKeys; i++) {
+			_dbg1("Add subkey " << i);
+			cbName = mex_key_length;
+			retCode = RegEnumKeyEx(hKey, i,
+				achKey,
+				&cbName,
+				nullptr,
+				nullptr,
+				nullptr,
+				&ftLastWriteTime);
+			if (retCode == ERROR_SUCCESS) {
+				ret.emplace_back(achKey); // Exception safety: strong guarantee (23.3.6.5, std::wstring is no-throw moveable 21.4.2)
```

Comment:
can we static assert on this? that "std::wstring is no-throw moveable 21.4.2)"

nice that we link to standard. please add standard version for clarity